File: src/main/java/scripting/shortcuts/CommandShortcut.java
package scripting.shortcuts;

import client.Character;
import client.Client;
import client.inventory.ItemVariationType;
import client.inventory.item.templates.ItemTemplate;
import client.inventory.slots.ItemSlot;
import client.inventory.slots.ItemSlotBundle;
import field.object.drop.EnterType;
import field.object.drop.ItemDrop;
import field.object.life.FieldMob;
import field.object.life.FieldNPC;
import lombok.Getter;
import lombok.Setter;
import managers.ItemManager;
import managers.MobManager;
import managers.NPCManager;
import managers.NPCShopManager;
import net.database.ShopAPI;
import net.maple.packets.CharacterPackets;
import net.server.Server;
import scripting.AbstractPlayerInteraction;

import static net.maple.handlers.user.UserChatHandler.refreshCommandList;

public class CommandShortcut extends AbstractPlayerInteraction {

    @Getter Client c;
    @Getter Character chr;
    @Getter @Setter String[] args;

    public CommandShortcut(Client c, String[] _args) {
        super(c);
        chr = c.getCharacter();
        args = _args;
    }

    public void gainMeso(int m) {
        chr.gainMeso(m);
    }

    public void reloadScripts(String script) {
        switch (script.toUpperCase()) {
            case "COMMANDS":
                refreshCommandList();
                break;
            default:
                break;
        }
    }

    public FieldNPC getNpc(int id) {
        return NPCManager.getNPC(id);
    }

    public FieldMob getMob(int id) {
        FieldMob mob = new FieldMob(MobManager.getMob(id), false);
        mob.setHp(mob.getTemplate().getMaxHP());
        mob.setMp(mob.getTemplate().getMaxMP());

        return mob;
    }

    public ItemTemplate getItemTemplate(int id) {
        ItemTemplate item = ItemManager.getItem(id);

        return item;
    }

    public void dropItem(int id, final int qty) {
        ItemSlot it = getItemTemplate(id).toItemSlot();

        if (it instanceof ItemSlotBundle) {
            ((ItemSlotBundle) it).setNumber((short) qty);
            ((ItemSlotBundle) it).setTitle(chr.getName());
        }

        ItemDrop drop = new ItemDrop(chr.getId(), chr, it, 0);
        drop.setPosition(chr.getPosition());
        chr.getField().enter(drop);
    }

    public void reloadShops() {
        Server.getInstance().setShops(ShopAPI.getShops());
        NPCShopManager.getInstance().reload();
    }

    public void addItem(int id, final int qty) {
        ItemTemplate item = ItemManager.getItem(id);
        if (item != null) {
            CharacterPackets.modifyInventory(chr,
                    i -> i.add(item, (short) qty),
                    false);
        }
    }

    public void reloadMap() {
        chr.getChannel().getFieldManager().reloadField(chr.getFieldId());
    }

    public void kickMe() {
        chr.getClient().close();
    }
}

File: src/main/java/scripting/command/CommandShortcut.kt
package scripting.command

import client.Character
import client.Client
import client.inventory.ModifyInventoriesContext
import client.inventory.item.templates.ItemTemplate
import client.inventory.item.slots.ItemSlotBundle
import field.obj.drop.ItemDrop
import field.obj.life.FieldMob
import field.obj.life.FieldNPC
import managers.ItemManager
import managers.NPCManager
import managers.NPCShopManager
import net.database.ShopAPI
import net.maple.handlers.user.UserChatHandler
import net.maple.packets.CharacterPackets.modifyInventory
import net.server.Server
import scripting.AbstractPlayerInteraction

class CommandShortcut(c: Client, _args: Array<String>) : AbstractPlayerInteraction(c) {

    val chr: Character = c.character
    val args: Array<String> = _args

    fun reloadScripts(script: String) {
        when (script.toUpperCase()) {
            "COMMANDS" -> UserChatHandler.refreshCommandList()
            else -> {
            }
        }
    }

    fun getNpc(id: Int): FieldNPC {
        return NPCManager.getNPC(id)
    }

    fun getMob(id: Int): FieldMob {
        val template = getMobTemplate(id)
        val mob = FieldMob(template, false)
        mob.hp = mob.template.maxHP
        mob.mp = mob.template.maxMP
        return mob
    }

    fun getItemTemplate(id: Int): ItemTemplate {
        return ItemManager.getItem(id)
    }

    fun dropItem(id: Int, qty: Int) {
        val template = getItemTemplate(id)
        val it = template.toItemSlot()
        if (it is ItemSlotBundle) {
            it.number = qty.toShort()
            it.title = chr.name
        }
        val drop = ItemDrop(chr.id, chr, it, 0)
        drop.position = chr.position
        chr.field.enter(drop)
    }

    fun reloadShops() {
        Server.shops = ShopAPI.shops
        NPCShopManager.reload()
    }

    fun addItem(id: Int, qty: Int) {
        val item = ItemManager.getItem(id)
        chr.modifyInventory({ i: ModifyInventoriesContext -> i.add(item, qty.toShort()) })
    }

    fun reloadMap() {
        chr.getChannel().fieldManager.reloadField(chr.fieldId)
    }

    fun kickMe() {
        chr.client.close()
    }
}